lib/bolt/logger.rb

```diff
@@ -105,6 +104,7 @@ def self.reset_logging
     end
 
     def self.warn_once(type, msg)
+      @mutex ||= Mutex.new
```

Comment:
Honestly i'm having a hard time understanding what instance variables do when there is a module with only class methods. I'm not sure if this is safe or not?

Reply:
This would be an instance variable of the module itself (which is an object). It will be shared between invocations of this method as you expect. It's not safe however, as this method could be called twice concurrently (the scenario the mutex is meant to guard against) and each caller could find `@mutex` uninitialized, create a Mutex and assign it.

I think the correct solution is probably to have the Mutex be a constant in the `Bolt::Logger` namespace that's initialized when the file is required.

Reply:
That is what I was afraid of. Thanks for the info.